### DISCUSSION TITLE
[RFE] Rulesets - Prevent Pull Requests to specific branches

### DISCUSSION THREAD

**Discussion by benipeled** [community/community#57404] [Repositories] [answered]
### Select Topic Area

Product Feedback

### Body

There are situations where we don't want to allow PRs to certain branches,
For example, all PRs should go through the dev branch and not directly to the master branch,

This RFE is about adding a new Rulesets option that will prevent the creation/edit of the PR target branch,

With such a rule - when users attempt to open a new PR or modify the target branch of an existing one, they will receive a notification (tooltip?) that the target branch is protected.

**Comment by patrick-knight** [comment, marked as answer]
Hey @benipeled are the [Restriction Creation](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#restrict-creations) and [Restrict updates](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#restrict-updates) rule suitable for this situation? They would prevent the creation of new branches or updates to existing branches that match the targeted branch name.

**Reply by benipeled** [reply]
Thanks @patrick-knight, indeed _Restrict updates_ did the trick:

![image](https://github.com/community/community/assets/31824530/11a1427c-83ae-4b04-9023-e4adb36b92ed)

I was initially confused about the 'Restrict updates' feature between branch content changes (creating a PR) and metadata changes (branch renaming). I believe we can improve the docs to avoid similar confusion in the future. Thanks anyway for the help!
